webapp/src/app.tsx

```diff
@@ -1,16 +1,81 @@
 import React, {useEffect} from 'react';
+import {useDispatch} from 'react-redux';
 
 import usePluginApi from 'src/hooks/usePluginApi';
 
 import Constants from 'src/plugin_constants';
 
+import {getGlobalModalState} from './selectors';
+import {setCurrentModalState} from './reducers/currentModal';
+import {resetGlobalModalState} from './reducers/globalModal';
+
 const GetConfig = (): JSX.Element => {
-    const {makeApiRequest} = usePluginApi();
+    const {makeApiRequest, pluginState, getApiState} = usePluginApi();
+    const {modalId, data} = getGlobalModalState(pluginState);
+    const dispatch = useDispatch();
+
+    const getConnectedUserState = () => {
+        const {isLoading, data: userData} = getApiState(Constants.pluginApiServiceConfigs.getConnectedUser.apiServiceName);
+        return {isLoading, data: userData as ConnectedState};
+    };
+
+    const getSubscriptionsConfiguredState = () => {
+        const {isLoading, isSuccess} = getApiState(Constants.pluginApiServiceConfigs.checkSubscriptionsConfigured.apiServiceName);
+        return {isLoading, isSuccess};
+    };
 
     useEffect(() => {
         makeApiRequest(Constants.pluginApiServiceConfigs.getConfig.apiServiceName);
     }, []);
 
+    useEffect(() => {
+        if (modalId) {
+            switch (modalId) {
+            case 'shareRecord':
+            case 'createIncident':
+                makeApiRequest(Constants.pluginApiServiceConfigs.getConnectedUser.apiServiceName);
+                break;
+            case 'addSubscription':
+            case 'editSubscription':
+                makeApiRequest(Constants.pluginApiServiceConfigs.checkSubscriptionsConfigured.apiServiceName);
+            }
+        }
+    }, [modalId]);
+
+    useEffect(() => {
+        const {data: userData, isLoading} = getConnectedUserState();
+        if (!isLoading && modalId) {
+            if (userData?.connected) {
+                switch (modalId) {
+                case 'shareRecord':
+                    dispatch(setCurrentModalState({modalId: 'shareRecord'}));
+                    break;
+                case 'createIncident':
+                    dispatch(setCurrentModalState({modalId: 'createIncident'}));
+                    break;
+                }
+            }
+            dispatch(resetGlobalModalState());
+        }
+    }, [getConnectedUserState().isLoading, getConnectedUserState().data]);
```

Comment:
Do we need `modalId` as a dependency here?

Reply:
no I don't think so, because we are making a request on a change of modal ID, which will change these states